apps/mymedicare_cb/authorization.py

```diff
@@ -59,127 +82,224 @@ def basic_auth(self):
+        This is used in the mymedicare_login() view at the start of the auth flow
+        and will produce an exception if not successful. This is also used by
+        the BB2 /health/external check.
+        """
+        headers = self.slsx_common_headers(request)
 
-    def user_signout(self, request):
-        headers = {"Authorization": "Bearer %s" % (self.auth_token)}
+        auth_flow_dict = get_session_auth_flow_trace(request)
 
-        # keep using deprecated conv - no conflict issue
-        headers.update({"X-SLS-starttime": str(datetime.datetime.utcnow())})
+        try:
+            response = requests.get(self.healthcheck_endpoint,
+                                    headers=headers,
+                                    allow_redirects=False,
+                                    verify=self.verify_ssl)
+            self.healthcheck_status_code = response.status_code
+            response.raise_for_status()
+        except (requests.exceptions.HTTPError, requests.exceptions.ConnectionError) as e:
+            if not called_from_health_external:
+                self.healthcheck_status_mesg = e
+                log_authenticate_start(auth_flow_dict, "FAIL",
+                                       "SLSx service health check error {reason}".format(reason=e),
+                                       slsx_client=self)
+            raise BBSLSxHealthCheckFailedException(settings.MEDICARE_ERROR_MSG)
 
-        if request is not None:
-            headers.update({"X-Request-ID": str(getattr(request, '_logging_uuid', None)
-                            if hasattr(request, '_logging_uuid') else '')})
+    def user_signout(self, request):
+        """
+        This uses the SLSx signout endpoint to sign out the bene.
+        After it does the signout, the response is a redirect to the
+        Medicare.gov signout page. Since this is more of a browser type
+        functionality, we disable the redirects in the HTTP GET call.
+        """
+        headers = self.slsx_common_headers(request)
+        headers.update(self.auth_header())
 
-        # Get auth flow session values.
         auth_flow_dict = get_session_auth_flow_trace(request)
 
         try:
             response = requests.get(self.signout_endpoint,
                                     headers=headers,
+                                    allow_redirects=False,
                                     verify=self.verify_ssl)
             self.signout_status_code = response.status_code
+            if self.signout_status_code != status.HTTP_302_FOUND:
+                log_authenticate_start(auth_flow_dict, "FAIL",
+                                       "SLSx signout response_code = {code}."
+                                       " Expecting HTTP_302_FOUND.".format(code=self.signout_status_code),
+                                       slsx_client=self)
+                raise BBMyMedicareSLSxSignoutException(settings.MEDICARE_ERROR_MSG)
             response.raise_for_status()
```

Comment:
Since allow_redirects is False, does response.raise_for_status() now throw an exception for HTTP 302 / 301s?  Just curious!